templates/common/infra/bicep/core/host/container-app.bicep

```diff
@@ -5,13 +5,19 @@ param tags object = {}
 param containerAppsEnvironmentName string
 param containerName string = 'main'
 param containerRegistryName string
+param secrets array = []
 param env array = []
 param external bool = true
 param imageName string
-param keyVaultName string = ''
-param managedIdentity bool = !empty(keyVaultName)
 param targetPort int = 80
 
+@description('Managed identity type')
+@allowed([ 'SystemAssigned', 'UserAssigned' ])
+param managedIdentityType string = 'SystemAssigned'
```

Comment:
I think this ends up meaning that by default, we would always assign a system identity. In [appservice.bicep](https://github.com/Azure/azure-dev/blob/6851f357ecd3a1ad74ef7aabd59d576f4cda8b61/templates/common/infra/bicep/core/host/appservice.bicep#L64), I don't think this is true. May be a good decision, but I don't yet know the implications here.

Reply:
Yes, this is correct.  When using defaults apps would use a `SystemAssigned` managed identity that would then be granted `AcrPull` role so we can out of the business of having to manage ACR secrets.